pkg/controllers/csi/provisioner/install.go

```diff
@@ -18,41 +18,28 @@ import (
 
 func (provisioner *OneAgentProvisioner) installAgentImage(
 	ctx context.Context,
-	dk dynakube.DynaKube,
+	dynakube dynakube.DynaKube,
 	latestProcessModuleConfig *dtclient.ProcessModuleConfig,
 ) (
-	targetImage string,
-	err error,
+	string,
+	error,
 ) {
-	tenantUUID, err := dk.TenantUUIDFromApiUrl()
+	tenantUUID, err := dynakube.TenantUUIDFromApiUrl()
 	if err != nil {
 		return "", err
 	}
 
-	targetImage = dk.CodeModulesImage()
+	targetImage := dynakube.CodeModulesImage()
 	// An image URI often contains one or several /-s, which is problematic when trying to use it as a folder name.
 	// Easiest to just base64 encode it
 	base64Image := base64.StdEncoding.EncodeToString([]byte(targetImage))
 	targetDir := provisioner.path.AgentSharedBinaryDirForAgent(base64Image)
-	targetConfigDir := provisioner.path.AgentConfigDir(tenantUUID, dk.GetName())
-
-	defer func() {
-		if err == nil {
-			err = processmoduleconfig.CreateAgentConfigDir(provisioner.fs, targetConfigDir, targetDir, latestProcessModuleConfig)
-		}
-	}()
```

Comment:
maybe this was a fix for a bug that we un-did by removing the `defer`?
(we do it now at https://github.com/Dynatrace/dynatrace-operator/pull/3604/files#diff-c506e4fd38e2f59453aa98279b508d2e5f6be2d46ca051c17a57252d42861e3bR62)

Reply:
Discussed it, this was needed because of the `codeModule, err := provisioner.db.ReadCodeModule(metadata.CodeModule{Version: targetImage})` returning early, we don't do this in the reverted state so its fine as is